frontend/src/app/settings/modelsources/t4c-settings/t4c-instance-settings/t4c-repo-deletion-dialog/t4c-repo-deletion-dialog.component.html

```diff
@@ -6,14 +6,41 @@
 <h1 mat-dialog-title>Remove T4C repository</h1>
 <div mat-dialog-content>
   <p>
-    Do you really want delete the T4C repository {{ repo.name }} with the ID
-    {{ repo.id }}?
+    Do you really want to delete the T4C repository
+    <span class="monospace">{{ repo.name }}</span
+    >? <br />
+    This will delete the repository in our database AND on the TeamForCapella
+    server.
   </p>
-  <p>The deletion cannot be undone!</p>
-</div>
-<div mat-dialog-actions>
-  <button mat-button (click)="this.dialogRef.close(false)">No</button>
-  <button mat-flat-button (click)="remoteRepository()">
-    Remove the repository
-  </button>
+  <p><b>The deletion cannot be undone!</b></p>
 </div>
+Please type in the name of the repository to confirm the deletion: <br />
+<form>
+  <mat-form-field appearance="fill">
+    <mat-label>Repository name</mat-label>
+    <input
+      (paste)="(false)"
+      matInput
+      [formControl]="repositoryNameForm"
+      [placeholder]="repo.name"
+    />
+    <mat-error *ngIf="repositoryNameForm.errors?.required">
+      You have to type in the repository name!
+    </mat-error>
+    <mat-error *ngIf="repositoryNameForm.errors?.repositoryNameMatchFailed">
+      The repository name doesn't match the repository you'd like to delete!
+    </mat-error>
+  </mat-form-field>
+
+  <div mat-dialog-actions class="flex-space-between">
+    <button mat-button (click)="this.dialogRef.close(false)">No</button>
+    <button
+      mat-flat-button
+      type="submit"
+      color="warn"
+      (click)="remoteRepository()"
+    >
```

Comment:
In order to bind the deletion to the form submission (as the `type="submit"` attribute on the `button` seems to hint), the function has to be bound to the `(submit)` attribute of the `form` instead of the `(click)` attribute of the `button`.

This enables among others the submission with the keyboard.

Reply:
Done.

Reply:
Still some issues of refreshing the list of repositories, but that’s strange and not a big deal. See https://github.com/DSD-DBS/capella-collab-manager/pull/304#issuecomment-1283566531

Reply:
Fixed in https://github.com/DSD-DBS/capella-collab-manager/pull/304/commits/79452514a87ddb7ce0d0d1da897905a71410f903